applications/CompressiblePotentialFlowApplication/custom_processes/define_3d_wake_process.cpp

```diff
@@ -363,32 +435,33 @@ void Define3DWakeProcess::CreateWakeSurfaceNodesAndElements(
     IndexType& rElement_index,
     const Properties::Pointer pElemProp) const
 {
-    const std::array<ModelPart::IndexType, 4> nodes_ids = CreateWakeSurfaceNodes(
+    KRATOS_TRY;
+    const std::array<ModelPart::IndexType, 4> nodes_ids = CreateWakeSurfaceNodes(ModelPart,
         rNode_index, rCoordinates1, rCoordinates2, rCoordinates3, rCoordinates4);
-
-    const double normal_projection = ComputeFaceNormalProjectionToWakeNormal(
-        rCoordinates1, rCoordinates2, rCoordinates3, rCoordinates4);
-
-    CreateWakeSurfaceElements(normal_projection, rElement_index, nodes_ids, pElemProp);
+        
+        const double normal_projection = ComputeFaceNormalProjectionToWakeNormal(
+            rCoordinates1, rCoordinates2, rCoordinates3, rCoordinates4);
+            
+            CreateWakeSurfaceElements(ModelPart, normal_projection, rElement_index, nodes_ids, pElemProp);
+    KRATOS_CATCH("");   
 }
 
 std::array<ModelPart::IndexType, 4> Define3DWakeProcess::CreateWakeSurfaceNodes(
+    ModelPart& ModelPart,
     IndexType& rNode_index,
     const array_1d<double, 3>& rCoordinates1,
     const array_1d<double, 3>& rCoordinates2,
     const array_1d<double, 3>& rCoordinates3,
     const array_1d<double, 3>& rCoordinates4) const
 {
-    const auto& p_node1 = mrStlWakeModelPart.CreateNewNode(
-        ++rNode_index, rCoordinates1[0], rCoordinates1[1], rCoordinates1[2]);
-    const auto& p_node2 = mrStlWakeModelPart.CreateNewNode(
-        ++rNode_index, rCoordinates2[0], rCoordinates2[1], rCoordinates2[2]);
-    const auto& p_node3 = mrStlWakeModelPart.CreateNewNode(
-        ++rNode_index, rCoordinates3[0], rCoordinates3[1], rCoordinates3[2]);
-    const auto& p_node4 = mrStlWakeModelPart.CreateNewNode(
-        ++rNode_index, rCoordinates4[0], rCoordinates4[1], rCoordinates4[2]);
-
+    KRATOS_TRY;
+    const auto& p_node1 = ModelPart.CreateNewNode(++rNode_index, rCoordinates1[0], rCoordinates1[1], rCoordinates1[2]);
+    const auto& p_node2 = ModelPart.CreateNewNode(++rNode_index, rCoordinates2[0], rCoordinates2[1], rCoordinates2[2]);
+    const auto& p_node3 = ModelPart.CreateNewNode(++rNode_index, rCoordinates3[0], rCoordinates3[1], rCoordinates3[2]);
+    const auto& p_node4 = ModelPart.CreateNewNode(++rNode_index, rCoordinates4[0], rCoordinates4[1], rCoordinates4[2]);
```

Comment:
```suggestion
    const auto p_node1 = ModelPart.CreateNewNode(++rNode_index, rCoordinates1[0], rCoordinates1[1], rCoordinates1[2]);
    const auto p_node2 = ModelPart.CreateNewNode(++rNode_index, rCoordinates2[0], rCoordinates2[1], rCoordinates2[2]);
    const auto p_node3 = ModelPart.CreateNewNode(++rNode_index, rCoordinates3[0], rCoordinates3[1], rCoordinates3[2]);
    const auto p_node4 = ModelPart.CreateNewNode(++rNode_index, rCoordinates4[0], rCoordinates4[1], rCoordinates4[2]);
```
Avoid pointer references.